src/oms/core/signer.ts

```diff
@@ -1,46 +1,193 @@
+    signedHeaders: string,
+    body: string
+}
+
+interface stringToSignParams {
+    iso8601: string,
+    yyyymmdd: string,
+    regionName: string,
+    serviceName: string,
+    hash: CryptoJS.lib.WordArray
+}
+
+interface signingKeyParams {
+    secretAccessKey: string,
+    dateStamp: string,
+    regionName: string,
+    serviceName: string
+}
+
+interface authHeaders {
+    /* eslint-disable */
+    'X-Sdk-Date': string,
+    'Authorization': string,
+    /* eslint-enable */
+}
+
+const SignAlgorithmHMACSHA256 = 'SDK-HMAC-SHA256'
+
+export class Signature {
+    getSignHeaders(credentials: credentialInfo, request: requestInfo, date: Date = new Date(), body = ''): authHeaders {
+        const dateFormat = /-|:|\..{3}/g
+        let currentDate = date.toISOString().replace(dateFormat, '')
+        if (!currentDate) {
+            currentDate = new Date().toISOString().replace(dateFormat, '')
+        }
+
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        const newHeaders = new Headers(request.headers)
+        newHeaders.set('Host', request.url.host)
+        newHeaders.append('X-Sdk-Date', currentDate)
+        const stringifiedHeaders = Signature.sortedStringifiedHeaders(newHeaders)
+        const signedHeaders = Signature.getSignedHeaders(stringifiedHeaders)
+        const { queryString, yyyymmdd } = Signature.getQueryString({
+            accessKeyId: credentials.accessKeyId,
+            regionName: credentials.regionName,
+            signedHeaders: signedHeaders,
+            serviceName: request.serviceName,
+            isoDate: currentDate
+        })
+        const { canonicalRequest, additionalQueryString } = Signature.getCanonicalRequest({
+            method: request.method,
+            url: request.url,
+            stringifiedHeaders: stringifiedHeaders,
+            signedHeaders: signedHeaders,
+            body: body
+        })
+        const hash = SHA256(canonicalRequest)
+        const stringToSign: string = Signature.getStringToSign({
+            iso8601: currentDate,
+            yyyymmdd: yyyymmdd,
+            regionName: credentials.regionName,
+            serviceName: request.serviceName,
+            hash: hash
+        })
+        const signatureKey = Signature.getSigningKey({
+            secretAccessKey: credentials.secretAccessKey,
+            dateStamp: yyyymmdd,
+            regionName: credentials.regionName,
+            serviceName: request.serviceName
+        })
+        const signature = Signature.getSignature(signatureKey, stringToSign)
+
+        return {
+            /* eslint-disable */
+            'X-Sdk-Date': currentDate,
+            'Authorization': `${queryString}${additionalQueryString} Signature=${signature}`,
+            /* eslint-enable */
+        }
+    }
+
+    /**
+     * Stringify and sort http headers
+     * @returns {Array<string>}
+     */
+    private static sortedStringifiedHeaders(headers: Headers): Array<string> {
```

Comment:
it's more convenient to work with the list because with list of string we do not need to do complex changes in other functions that using result of sortedHeaders